examples/vector_fe/vector_fe_ex5/vector_fe_ex5.C

```diff
@@ -133,7 +137,24 @@ main(int argc, char ** argv)
   nl_system.solve();
 
 #ifdef LIBMESH_HAVE_EXODUS_API
-  ExodusII_IO(mesh).write_equation_systems("out.e", equation_systems);
+  if (! approx_order) /* CONSTANT, MONOMIAL_VEC */
+    {
+      // Warn about trivial solution for CONSTANT approximations (Poisson must be at least C0)
+      libMesh::out << "\nWARNING: The elemental vector order is CONSTANT. The solution" << std::endl
+                   << "written out to 'out_constant.e' will be trivial." << std::endl;
+
+      // Need to get string of variable names (vector components) for 'set_output_variables()'
+      std::vector<std::string> varnames;
+      equation_systems.build_variable_names(varnames);
+
+      // Create ExodusII object with an unambigous filename (indicating this solution is special)
+      ExodusII_IO exo_ptr(mesh);
+      exo_ptr.write("out_constant.e");
+      exo_ptr.set_output_variables(varnames);
+      exo_ptr.write_element_data(equation_systems);
+    }
+  else
+    ExodusII_IO(mesh).write_equation_systems("out.e", equation_systems);
```

Comment:
I live behind the MOOSE wall by and large. Do we really not handle element vs. nodal writing internally? It seems like in an ideal world there would be no if/else branch here. (Other than to warn about the trivial solution)

Reply:
@lindsayad the `write_equation_systems()` method used here points to the namespace in `mesh/mesh_output.h`, which basically proceeds through the same steps I hard-coded here but for nodal data only (or element data written as nodal). I think this method ought to be made to handle constant element data as well, and I tried to do so, but it was tricky and I ultimately gave up on that.

Reply:
Yeah, the API design in libMesh here is garbage.  Not my garbage, but I'm not going to name names; I'm certainly responsible for letting it slide at the time rather than insisting that write_equation_systems() be upgraded instead.  I think that a decade ago I still thought of Exodus as a viz format rather than as something we wanted to make into a first class restart option someday.

If the higher-order support going into Exodus (which is what limited my respect for it in the past; I couldn't even properly visualize my own dissertation work without adding fake h refinement to make up for the low p) is usable enough (and not just limited to use in conjunction with the new spline support) then we might have an opportunity to fix all this soon.

For now?  @crswong888 reiterated to me that we can just drop the vector_fe_ex5 commit, but I think the ugly API here makes it more worth leaving in, for didactic reasons, until we make Exodus `write_equation_systems()` more versatile.